syntax = "proto2";

message EuropeanBankEmpty {}

message EuropeanBankIdRequest {
  required string id = 1;
}

message EuropeanBankNotIdRequest {
  required string bankName = 1;
  required string fullName = 2;
  required string swift = 3;
  required string iban = 4;
  optional string currency = 5;
  required string numberAccount = 6;

  enum Type {
    saving = 0;
    current = 1;
  }

  required Type type = 7;
  required string bankAddress = 8;
  required string userAddress = 9;  
  required string documentIdentification = 10;
}

message EuropeanBankRequest {
  required string id = 1; 
  required string bankName = 2;
  required string fullName = 3;
  required string swift = 4;
  required string iban = 5;
  optional string currency = 6;
  required string numberAccount = 7;

  enum Type {
    saving = 0;
    current = 1;
  }

  required Type type = 8;
  required string bankAddress = 9;
  required string userAddress = 10;  
  required string documentIdentification = 11;
}

message EuropeanBanksTableRequest {
  required int32 page = 1;
  optional int32 per_page = 2 [default = 15];
  optional string search = 3;
}

message EuropeanBanksResponse {
  required EuropeanBankRequest european = 1;
}

message EuropeanBanksMultipleResponse {
  repeated EuropeanBankRequest european = 1;
}

message EuropeanBanksTableResponse {
  repeated EuropeanBankRequest items = 1;
  required int32 page = 2;
  required int32 per_page = 3;
  required int32 total_items = 4;
  required int32 num_pages = 5;
}


service EuropeanBanks {
  rpc table (EuropeanBanksTableRequest) returns (EuropeanBanksTableResponse);
  rpc get_all (EuropeanBankEmpty) returns (EuropeanBanksMultipleResponse);
  rpc get (EuropeanBankIdRequest) returns (EuropeanBanksResponse);
  rpc save (EuropeanBankNotIdRequest) returns (EuropeanBanksResponse);
  rpc update (EuropeanBankRequest) returns (EuropeanBanksResponse);
  rpc delete (EuropeanBankIdRequest) returns (EuropeanBanksResponse);
}

